libsolidity/codegen/ir/IRGenerator.cpp

```diff
@@ -128,6 +130,9 @@ void IRGenerator::generateQueuedFunctions()
 
 string IRGenerator::generateFunction(FunctionDefinition const& _function)
 {
+	if (_function.isConstructor())
```

Comment:
This looks dangerous. Is it possible to either generate the code for the constructor or create an assertion?